File: javascript/ql/src/experimental/Security/CWE-918/SSRF.js
const axios = require('axios');

export const handler = async (req, res, next) => {
  const { target } = req.body;

  try {
    // BAD: `target` is controlled by the attacker
    const response = await axios.get('https://example.com/current_api/' + target);

    // process request response
    use(response);
  } catch (err) {
    // process error
  }
};

File: go/ql/src/experimental/CWE-918/SSRF.go
package main

import (
	"net/http"
)

func handler(w http.ResponseWriter, req *http.Request) {
	target := req.FormValue("target")

	// BAD: `target` is controlled by the attacker
	resp, err := http.Get("https://example.com/current_api/" + target)
	if err != nil {
		// error handling
	}

	// process request response
	use(resp)
}
